frame/support/procedural/src/construct_runtime/expand/inherent.rs

```diff
@@ -162,12 +162,12 @@ pub fn expand_outer_inherent(
 		}
 
 		impl #scrate::traits::EnsureInherentsAreFirst<#block> for #runtime {
-			fn ensure_inherents_are_first(block: &#block) -> Result<(), u32> {
+			fn ensure_inherents_are_first(block: &#block) -> Result<u32, u32> {
 				use #scrate::inherent::ProvideInherent;
 				use #scrate::traits::{IsSubType, ExtrinsicCall};
 				use #scrate::sp_runtime::traits::Block as _;
 
-				let mut first_signed_observed = false;
+				let mut first_signed_observed = 0;
```

Comment:
Is it ok if this is never incremented? i.e. there are no signed

Reply:
looking at `execute_extrinsics_with_book_keeping` if there are no signed we should return something like `None` here and then in  `execute_extrinsics_with_book_keeping` make sure to call `on_post_inherent` after iterating over the extrinsics.

Reply:
yeah, then the first index in the extrinsics vector is the signed extrinsic, so we would then immediately execute any `on_post_inherent` right after `on_initialize` and before any extrinsics.